core/include/scipp/core/transform_subspan.h

```diff
@@ -22,12 +22,12 @@ static constexpr auto need_subspan = [](const VariableConstProxy &var,
 
 static constexpr auto maybe_subspan = [](VariableConstProxy &var,
                                          const Dim dim) {
-  Variable ret;
+  auto ret = std::make_unique<Variable>();
   if (need_subspan(var, dim)) {
-    ret = subspan_view(var, dim);
-    var = ret;
+    *ret = subspan_view(var, dim);
+    var = *ret;
   }
-  return ret;
+  return std::move(ret);
```

Comment:
Is `keep_subspan_vars_alive` in line 52 below needed now?

Reply:
I think so, it's the one that will keep/destroy the `unique_ptr`s in that scope